06_gpu_and_ml/unsloth_finetune.py

```diff
@@ -0,0 +1,562 @@
+# ---
+# cmd: ["modal", "run", "-m", "06_gpu_and_ml.unsloth_finetune"]
+# args: ["--max-steps", "1", "--save-steps", "1", "--skip-eval"]
+# ---
+
+# # Efficient LLM Finetuning with Unsloth
+
+# Training large language models is an incredibly compute-hungry process.
+# Open-source LLMs often require many GBs (or in extreme cases,
+# [one TB](https://github.com/MoonshotAI/Kimi-K2#2-model-summary)!) of
+# VRAM just to fit in memory. Finetuning models requires even more memory;
+# a common estimate for naive finetuning puts the VRAM requirements at roughly
+# 4.2x the original model size:
+# 1x for model weights + 1x for gradients + 2x for optimizer state + 20% for activations.
+# Parameter efficient methods like LoRA can improve matters significantly, since
+# this estimate now applies to just the LoRA modules' weights, rather than the entire
+# model's. Further gains can be made with quantization of each of the components mentioned
+# above, but doing so requires quantization-aware training, which can be tricky to
+# combine with methods like LoRA.
+
+# Unsloth provides optimized methods for LLM finetuning with LoRA and quantization,
+# leading to typical performance gains of 2x faster training with 70% less memory usage.
+# This example demonstrates using Unsloth to finetune a version of Qwen3-14B with the
+# FineTome-100k dataset on Modal using only a single GPU!
+
+# To run this example, checkout the examples repo, install the Modal client, and run
```

Comment:
This info shows up in the sidebar in the docs, so I wouldn't put it here.

Also, users tend to copy-paste code segment by segment, so it's better to put the command down where it's runnable (or runnable once they also copy the utilities/addenda).

So I think it's better to put this info above the local entrypoint in triple-backticks, i.e. ```bash.